bin/github-review.sh

```diff
@@ -0,0 +1,313 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+usage() {
+    cat <<EOF
+This script verifies the deployed contracts on a chain ID for a given PR.
+
+USAGE
+    bash ./bin/github-review.sh <PR> <CHAIN_ID> <RPC_URL> <VERSION>
+
+ARGUMENTS
+    PR          The GitHub PR number
+    CHAIN_ID    The chain ID to verify
+    RPC_URL     The RPC URL to use for the chain ID
+    VERSION     The version of the contracts to verify
+
+EXAMPLES
+    bash ./bin/github-review.sh 123 1 https://rpc.ankr.com/eth 1.3.0
+    bash ./bin/github-review.sh 123 1 https://rpc.ankr.com/eth 1.4.1
+EOF
+}
+
+if [[ -n "$(git status --porcelain)" ]]; then
+    echo "ERROR: Dirty Git index, please commit all changes before continuing" 1>&2
+    exit 1
+fi
+if ! command -v gh &> /dev/null; then
+    echo "ERROR: Please install the 'gh' GitHub CLI" 1>&2
+    exit 1
+fi
+if ! command -v cast &> /dev/null; then
+    echo "ERROR: Please install the 'cast' Foundry (forge)" 1>&2
+    exit 1
+fi
+
+pr=0
+chainid=0
+rpc=0
+version=""
+# TODO: "1.0.0" "1.1.1" "1.2.0"
+availableVersion=("1.3.0" "1.4.1")
+case $# in
+    4)
+        if ! [[ $1 =~ ^[0-9]+$ ]]; then
+            echo "ERROR: $1 is not a valid GitHub PR number" 1>&2
+            usage
+            exit 1
+        fi
+        pr=$1
+        if ! [[ $2 =~ ^[0-9]+$ ]]; then
+            echo "ERROR: $2 is not a valid Chain ID number" 1>&2
+            usage
+            exit 1
+        fi
+        chainid="$(cast chain-id --rpc-url $3)"
+        if [[ $chainid != $2 ]]; then
+            echo "ERROR: RPC $3 doesn't match chain ID $2" 1>&2
+            usage
+            exit 1
+        fi
+        rpc=$3
+        if [[ ! " ${availableVersion[@]} " =~ " ${4} " ]]; then
```

Comment:
This is a cleaver way for checking list inclusion! (For other less Bash-y ppl, `=~` is regex match). I wonder if `1.4_1` works (i.e. is `.` the regex "any character" or an actual '.')?

Reply:
Copilot: `" ${4} "`: This is the fourth argument passed to the script, surrounded by spaces and quotes to form a regular expression. The spaces ensure that the match must be a whole word. For example, if `${4}` is `2`, it will match `2` in the array, but not `20` or `12`.

Reply:
TIL, from the Bash manual:

>  If any part of the pattern is quoted, the quoted portion is matched literally.  This means every character in the quoted portion matches itself, instead of having any special pattern matching meaning. If the pattern is stored in a shell variable, quoting the variable expansion forces the entire pattern to be matched literally.  Treat bracket expressions in regular expressions carefully, since normal quoting and pattern characters lose their meanings between brackets.

So, the fact that the expression is quotes makes it be matched literally and NOT like a regular expression (contrary to what I understood from Copilot). For example:

```
$ [[ "10" =~ 1. ]] && echo true || echo false
true
$ [[ "10" =~ "1." ]] && echo true || echo false
false
$ pattern='1.'
$ [[ "10" =~ $pattern ]] && echo true || echo false
true
$ [[ "10" =~ "$pattern" ]] && echo true || echo false
false
```

Bash is SO weird...